### PR TITLE
Emad/refactor update controlling member

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

- added OrganizationId for using in crating dynamic task (update. request)
- modified task phrases (subject, description) and configs based on confluence
Note: decision needed about the `LicenceId` property in task creation

### CODE REVIEW

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 1]
```diff
@@ -280,20 +280,46 @@ private async Task<ChangeSpec> AddDynamicTasks(ControllingMemberCrcApplicationRe
         CancellationToken ct)
     {
         ChangeSpec changes = new();
+        //PeaceOfficerStatusChanged: check if Hold a Position with Peace Officer Status changed, create task with high priority, assign to Licensing CS team
+        PoliceOfficerRoleCode? originalRoleCode = originalApp.PoliceOfficerRoleCode == null ? null
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:284]: it should compare with contact, not originalApp

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 2]
```diff
@@ -280,20 +280,46 @@ private async Task<ChangeSpec> AddDynamicTasks(ControllingMemberCrcApplicationRe
...
+        if (newRequest.IsPoliceOrPeaceOfficer != originalApp.IsPoliceOrPeaceOfficer ||
+            newRequest.PoliceOfficerRoleCode != originalRoleCode ||
+            newRequest.OtherOfficerRole != originalApp.OtherOfficerRole ||
+            newFileInfos.Any(d => d.LicenceDocumentTypeCode == LicenceDocumentTypeCode.PoliceBackgroundLetterOfNoConflict))
+        {
+            IEnumerable<string> fileNames = newFileInfos.Where(d => d.LicenceDocumentTypeCode == LicenceDocumentTypeCode.PoliceBackgroundLetterOfNoConflict).Select(d => d.FileName);
+            changes.PeaceOfficerStatusChanged = true;
+            changes.PeaceOfficerStatusChangeTaskId = (await _taskRepository.ManageAsync(new CreateTaskCmd()
+            {
+                Description = $"Business have to requested to update below submitted information for its controlling member:" +
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:296]: grammer error in this sentence

Comment (peggy-quartech) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:296]: I still suggest you to ask Kenny to write specific cm update task wording.

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 3]
```diff
@@ -280,20 +280,46 @@ private async Task<ChangeSpec> AddDynamicTasks(ControllingMemberCrcApplicationRe
...
+                $"\nControlling Member - {originalApp.Surname + ", " + originalApp.GivenName}\nUpdated: Peace Officer Status\nFile Names {string.Join("\n* ", fileNames)} ",
+                DueDateTime = DateTimeOffset.Now.AddDays(3),
+                Subject = $"Updating controlling member background information for Case Number {originalApp.CaseNumber}",
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:299]: what case number? controlling member application case number or biz app case number?

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 4]
```diff
@@ -280,20 +280,46 @@ private async Task<ChangeSpec> AddDynamicTasks(ControllingMemberCrcApplicationRe
...
+                TaskPriorityEnum = TaskPriorityEnum.Normal,
+                RegardingAccountId = originalApp.OrganizationId,
+                AssignedTeamId = Guid.Parse(DynamicsConstants.Licensing_Risk_Assessment_Coordinator_Team_Guid),
+                //TODO: in confluence it says licence, but don't have licence in crc app, currently is filling with null (it should be a Guid, doesn't apply CaseNumber)
+                LicenceId = null,
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:304]: ask kenny

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 5]
```diff
@@ -280,20 +280,46 @@ private async Task<ChangeSpec> AddDynamicTasks(ControllingMemberCrcApplicationRe
...
             changes.MentalHealthStatusChanged = true;
             IEnumerable<string> fileNames = newFileInfos.Where(d => d.LicenceDocumentTypeCode == LicenceDocumentTypeCode.MentalHealthCondition).Select(d => d.FileName);
             changes.MentalHealthStatusChangeTaskId = (await _taskRepository.ManageAsync(new CreateTaskCmd()
             {
-                Description = $"Please see the attached mental health condition form submitted by the controlling member : {string.Join(";", fileNames)}  ",
+                Description = $"Business have to requested to update below submitted information for its controlling member:" +
+                $"\nControlling Member - {originalApp.Surname + ", " + originalApp.GivenName}\nUpdated: Mental Health\nFile Names {string.Join("\n* ", fileNames)} ",
                 DueDateTime = DateTimeOffset.Now.AddDays(3),
-                Subject = $"Mental Health Condition Update on {originalApp.CaseNumber}",
+                Subject = $"Updating controlling member background information for {originalApp.CaseNumber}",
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:317]: same question as above

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 6]
```diff
@@ -303,13 +329,15 @@ private async Task<ChangeSpec> AddDynamicTasks(ControllingMemberCrcApplicationRe
...
             changes.CriminalHistoryStatusChangeTaskId = (await _taskRepository.ManageAsync(new CreateTaskCmd()
             {
-                Description = $"Please see the criminal charges submitted by the licensee with details as following : {newRequest.CriminalHistoryDetail}",
+                Description = $"Business have to requested to update below submitted information for its controlling member:" +
+                $"\nControlling Member - {originalApp.Surname + ", " + originalApp.GivenName}\nUpdated: Criminal History with details:\n{newRequest.CriminalHistoryDetail} ",
                 DueDateTime = DateTimeOffset.Now.AddDays(3), //will change when dynamics agree to calculate biz days on their side.
-                Subject = $"Criminal Charges or New Conviction Update on {originalApp.CaseNumber}",
-                TaskPriorityEnum = TaskPriorityEnum.High,
-                RegardingContactId = originalApp.ContactId,
+                Subject = $"Updating controlling member background information for {originalApp.CaseNumber}",
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:335]: same question

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:1]: controlling member update does not create/update any application
Please remove the corresponding lines

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs (hunk 2 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:296]: I copied from confluence, accept that it is not right, I will ask Kenny

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs (hunk 3 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:299]: it is controlling member, would ask Kenny.

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs (hunk 1 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs]: replaced to compare with contact

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs (hunk 2 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs]: checked the task creation with Kenny, new modified function should be ok

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 7]
```diff
@@ -77,43 +81,49 @@ public async Task<ControllingMemberCrcAppResponse> Handle(GetControllingMemberCr
...
-        updateCmd.Id = originalApp.ContactId ?? Guid.Empty;
+        updateCmd.Id = (Guid) request.ApplicantId;
         await _contactRepository.ManageAsync(updateCmd, ct);
 
+        //TODO: check the required parameters to give in function upload
         await UploadNewDocsAsync(request.DocumentExpiredInfos,
             cmd.LicAppFileInfos,
-            response.ControllingMemberAppId,
-            response.ContactId,
+            //set link to applicant only, applicant Id should be null
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:114]: //set link to applicant only, application Id should be null

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 8]
```diff
@@ -274,44 +284,89 @@ await _cmInviteRepository.ManageAsync(
...
+                              $"\nControlling Member - {originalApplicant.LastName}, {originalApplicant.FirstName}" +
+                              $"\nUpdated:\n{descriptionBuilder.ToString().Trim()}" +
+                              $"{(fileNames.Any() ? $"\n\nFile Names:\n* {string.Join("\n* ", fileNames)}" : string.Empty)}",
+                DueDateTime = DateTimeOffset.Now.AddDays(3),
+                Subject = $"Updating controlling member background information for {bizLicApp.CaseNumber}",
+                TaskPriorityEnum = TaskPriorityEnum.Normal,
+                RegardingAccountId = organizationId,
                 AssignedTeamId = Guid.Parse(DynamicsConstants.Licensing_Risk_Assessment_Coordinator_Team_Guid),
-                LicenceId = originalApp.ControllingMemberAppId
+                LicenceId = bizLicApp.LicenceAppId
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:352]: is this licenceId or licence appId?

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs (hunk 8 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:352]: it should be Business License Id, I was thinking it is bizLicApp, thanks for noticing that

Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:352]: Licence Id

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**